File: abc-270/c.py
import sys

sys.setrecursionlimit(10 ** 9)

N, X, Y = map(int, input().split())
Branches = [[] for _ in range(N)]
Seen = [False for _ in range(N)]
Simple_Path = []


def inputBranches() -> None:
    for _ in range(N - 1):
        u, v = map(int, input().split())
        Branches[u - 1].append(v - 1)
        Branches[v - 1].append(u - 1)


def dfs(x: int, y: int) -> bool:
    Seen[x] = True
    if y in Branches[x]:
        Simple_Path.append(y)
        return True

    for next in Branches[x]:
        if Seen[next]:
            continue

        if dfs(next, y):
            Simple_Path.append(next)
            return True
    return False


def print_simple_path() -> None:
    str_list = [str(route + 1) for route in Simple_Path]
    print(' '.join(str_list))


def main():
    inputBranches()
    dfs(Y - 1, X - 1)
    Simple_Path.append(Y - 1)
    print_simple_path()


if __name__ == '__main__':
    main()

File: abc-270/c.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

var sc = bufio.NewScanner(os.Stdin)
var rdr = bufio.NewReaderSize(os.Stdin, 1000000)

func stringInput() string {
	sc.Scan()
	return sc.Text()
}

func intInput() int {
	sc.Scan()
	x, err := strconv.Atoi(sc.Text())
	if err != nil {
		panic(err)
	}
	return x
}

func intSliceInput(n int) []int {
	sl := []int{}
	for i := 0; i < n; i++ {
		x := intInput()
		sl = append(sl, x)
	}
	return sl
}

func min(x, y int) int {
	if x < y {
		return x
	}
	return y
}

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

func initSlSl(y, x int) [][]int {
	sl := make([][]int, y)
	if x == -1 {
		for i := 0; i < y; i++ {
			sl[i] = []int{}
		}
		return sl
	}
	for i := 0; i < y; i++ {
		sl[i] = make([]int, x)
	}
	return sl
}

func init() {
	sc.Split(bufio.ScanWords)
}

var simplePath = []int{}
var seen = []bool{}

func contains(sl []int, x int) bool {
	for _, v := range sl {
		if v == x {
			return true
		}
	}
	return false
}

func initSeen(n int) {
	for i := 0; i < n; i++ {
		seen = append(seen, false)
	}
}

func bfs(branches [][]int, x, y int) bool {
	seen[x] = true
	if contains(branches[x], y) {
		simplePath = append(simplePath, y)
		return true
	}
	for _, next := range branches[x] {
		if seen[next] {
			continue
		}
		if bfs(branches, next, y) {
			simplePath = append(simplePath, next)
			return true
		}
	}
	return false
}

func printIntSlice(sl []int) {
	ans := make([]string, 0)
	for _, v := range sl {
		ans = append(ans, strconv.Itoa(v + 1))
	}
	fmt.Println(strings.Join(ans, " "))
}

func main() {
	n, x, y := intInput(), intInput()-1, intInput()-1
	branches := make([][]int, n)

	initSeen(n)
	for i := 0; i < n-1; i++ {
		u, v := intInput()-1, intInput()-1
		branches[u] = append(branches[u], v)
		branches[v] = append(branches[v], u)
	}
	bfs(branches, y, x)
	simplePath = append(simplePath, y)
	printIntSlice(simplePath)

}
